cmd/drop.rb

```diff
@@ -81,13 +81,24 @@ def drop
                 lines = []
                 has_removed_line = false
 
+                # Store comments that look like they were generated by `--describe` in a buffer.
+                # If it turns out the comments are directly preceding a line that should be dropped,
+                # drop the comments retroactively by resetting the array without pushing it to `lines`.
+                comment_line_buffer = []
+
                 File.foreach(brewfile) do |line|
-                    unless line.match(/^\s*#{brewfile_prefix_type}\s+["'](#{regex_name})["']/)
-                        lines.push(line)
+                    if line.match(/^\s*# (?!brew|cask|tap|mas|whalebrew|vscode)\w/)
```

Comment:
Just curious, why bother with this `?!` match here? To avoid capturing places where someone commented out a line like `# brew "foo"` ?

Reply:
Yes, that's the goal. It's a pretty ugly hack but I think it'll address the most common cases where a comment would otherwise incorrectly be removed.